exercises/linked-list/canonical-data.json

```diff
@@ -0,0 +1,543 @@
+          {
+            "operation": "push",
+            "value": 71
+          },
+          {
+            "operation": "push",
+            "value": 83
+          },
+          {
+            "operation": "push",
+            "value": 79
+          },
+          {
+            "operation": "delete",
+            "value": 83
+          },
+          {
+            "operation": "count",
+            "expected": 2
+          },
+          {
+            "operation": "pop",
+            "expected": 79
+          },
+          {
+            "operation": "shift",
+            "expected": 71
+          }
+        ]
+      },
+      "expected": {}
+    },
+    {
+      "uuid": "ee3729ee-3405-4bd2-9bad-de0d4aa5d647",
+      "description": "deletes the first of two elements",
+      "property": "list",
+      "input": {
+        "operations": [
+          {
+            "operation": "push",
+            "value": 97
+          },
+          {
+            "operation": "push",
+            "value": 101
+          },
+          {
+            "operation": "delete",
+            "value": 97
+          },
+          {
+            "operation": "count",
+            "expected": 1
+          },
+          {
+            "operation": "pop",
+            "expected": 101
+          }
+        ]
+      },
+      "expected": {}
+    },
+    {
+      "uuid": "47e3b3b4-b82c-4c23-8c1a-ceb9b17cb9fb",
+      "description": "deletes the second of two elements",
+      "property": "list",
+      "input": {
+        "operations": [
+          {
+            "operation": "push",
+            "value": 97
+          },
+          {
+            "operation": "push",
+            "value": 101
+          },
+          {
+            "operation": "delete",
+            "value": 101
+          },
+          {
+            "operation": "count",
+            "expected": 1
+          },
+          {
+            "operation": "pop",
+            "expected": 97
+          }
+        ]
+      },
+      "expected": {}
+    },
+    {
+      "uuid": "7b420958-f285-4922-b8f9-10d9dcab5179",
+      "description": "delete does not modify the list if the element is not found",
+      "property": "list",
+      "input": {
+        "operations": [
+          {
+            "operation": "push",
+            "value": 89
+          },
+          {
+            "operation": "delete",
+            "value": 103
+          },
+          {
+            "operation": "count",
+            "expected": 1
+          }
+        ]
+      },
+      "expected": {}
+    },
+    {
+      "uuid": "7e04828f-6082-44e3-a059-201c63252a76",
+      "description": "deletes only the first occurence",
+      "property": "list",
+      "input": {
+        "operations": [
+          {
+            "operation": "push",
+            "value": 73
+          },
+          {
+            "operation": "push",
+            "value": 9
+          },
+          {
+            "operation": "push",
+            "value": 9
```

Comment:
it strikes me that because these two 9 are next to each other, it's not defined nor tested whether the element closer to the head or to the tail should be deleted. but perhaps that is intentional.

Reply:
That was intentional; it's not in the description and wanted to "leave it in the middle". Do yo think it's better to _enforce_ it? Or maybe I should add a comment that it's left to the implementor?

Reply:
Right now I don't really think it increases educational value to enforce one way or the other. Probably a comment that it's left up to dealer's choice would pre-emptively answer anyone else who asks a similar question as I did.

Reply:
I'll add a helpful comment :)